### PR TITLE
refactor: decode user in storage

### PR DESCRIPTION
**Pull request by n1ru4l**
### Background

I will touch this code soon, so I wanted ti first clean it up a bit.

### Checklist

- [ ] Input validation
- [ ] Output encoding
- [ ] Authentication management
- [ ] Session management
- [ ] Access control
- [ ] Cryptographic practices
- [ ] Error handling and logging
- [ ] Data protection
- [ ] Communication security
- [ ] System configuration
- [ ] Database security
- [ ] File management
- [ ] Memory management
- [ ] Testing

### CODE REVIEW

File: packages/services/api/src/modules/schema/providers/schema-manager.ts [hunk 1]
```diff
@@ -969,9 +969,8 @@ export class SchemaManager {
       return null;
     }
 
-    return this.storage.getOrganizationUser({
-      organizationId: args.organizationId,
-      userId: args.userId,
+    return this.storage.getUserById({
+      id: args.userId,
```
Comment (n1ru4l) [inline packages/services/api/src/modules/schema/providers/schema-manager.ts:973]: Method was only used here and did not provide a real benefit over the exiting one..

File: packages/services/api/src/modules/shared/providers/storage.ts [hunk 2]
```diff
@@ -69,7 +69,6 @@ export interface Storage {
   isReady(): Promise<boolean>;
   ensureUserExists(_: {
     superTokensUserId: string;
-    externalAuthUserId?: string | null;
```
Comment (n1ru4l) [inline packages/services/api/src/modules/shared/providers/storage.ts:72]: We do not use the externalAuthUserId anywhere for a long time now.

Comment (dotansimha) [reply packages/services/api/src/modules/shared/providers/storage.ts:72]: This was used with Auth0, it's legacy, and can be removed for sure.

File: packages/services/storage/src/index.ts [hunk 3]
```diff
@@ -477,64 +426,55 @@ export async function createStorage(
...
         email: string;
         fullName: string;
         displayName: string;
-        externalAuthUserId: string | null;
         oidcIntegrationId: string | null;
       },
       connection: Connection,
     ) {
-      const user = await connection.one<users>(
+      await connection.query<unknown>(
```
Comment (dotansimha) [inline packages/services/storage/src/index.ts:463]: why `unknown`? 🤔

File: packages/services/storage/src/index.ts [hunk 4]
```diff
@@ -477,64 +426,55 @@ export async function createStorage(
...
       `);
 
-      if (user) {
-        return transformUser(user);
+      if (!record) {
+        return null;
       }
 
-      return null;
+      return UserModel.parse(record);
```
Comment (dotansimha) [inline packages/services/storage/src/index.ts:445]: thank you 🙇

**Review by dotansimha — APPROVED** [review]

File: packages/services/storage/src/index.ts (hunk 3 above)
Comment (n1ru4l) [reply packages/services/storage/src/index.ts:463]: because we don't want anyone to use that 🗡️

**Pull request merged**